import "libnutrition/proto/data/unit.proto";
import "libnutrition/proto/service/error.proto";

message UnitLoadRequest
{
	optional bool omit = 1 [default = false];
	optional bool all = 2 [default = false];
	
	enum Dimension {
		Weight = 0;
        Volume = 1;
        Quantity = 2;
        Serving = 3;
        Energy = 4;
        IU = 5;
    }
    
    repeated Dimension requestedDimensions = 3;
    
    repeated string requestedAbbreviations = 4;
    
    optional bool basicUnitsOnly = 5 [default = false];
}

message UnitLoadResponse 
{
	repeated UnitData units = 1;
	optional Error error = 100;
}